scripts/saveImage.sh

```diff
@@ -0,0 +1,202 @@
+# Normally at least the exposure will be passed and the sensor temp if known.
+while [ $# -gt 0 ]; do
+	VARIABLE="THIS_${1%=*}"		# everything before the "="
+	VALUE="${1##*=}"			# everything after the "="
+	shift
+	# Export the variable so other scripts we call can use it.
+	export ${VARIABLE}="${VALUE}"	# need "export" to get indirection to work
+done
+
+# Export so other scripts can use it.
+export CURRENT_IMAGE="${IMAGE_TO_USE}"		# darkCapture.sh and darkSubtract.sh expect CURRENT_IMAGE
+source "${ALLSKY_SCRIPTS}/darkCapture.sh"		# does not return if in darkframe mode
+# xxxx TODO: Dark subtract long-exposure images, even if during daytime.
+# xxxx TODO: Need a config variable to specify the threshold to dark subtract.
+# xxxx TODO: Possibly also for stretching below.
+if [ "${DAY_OR_NIGHT}" = "NIGHT" ] ; then
+	source "${ALLSKY_SCRIPTS}/darkSubtract.sh"	# It will modify the image but not its name.
+fi
+
+# If any of the "convert"s below fail, exit since we won't know if the file was corrupted.
+
+# Resize the image if required
+if [ "${IMG_RESIZE}" = "true" ] ; then
+	[ "${ALLSKY_DEBUG_LEVEL}" -ge 4 ] && echo "${ME}: Resizing '${IMAGE_TO_USE}' to ${IMG_WIDTH}x${IMG_HEIGHT}"
+	convert "${IMAGE_TO_USE}" -resize "${IMG_WIDTH}x${IMG_HEIGHT}" "${IMAGE_TO_USE}"
+	if [ $? -ne 0 ] ; then
+		echo -e "${RED}*** ${ME}: ERROR: IMG_RESIZE failed; not saving{$NC}"
+		exit 4
+	fi
+fi
+
+# Crop the image if required
+if [ "${CROP_IMAGE}" = "true" ] ; then
+	[ "${ALLSKY_DEBUG_LEVEL}" -ge 4 ] && echo "${ME}: Cropping '${IMAGE_TO_USE}' to ${CROP_WIDTH}x${CROP_HEIGHT}"
+	convert "${IMAGE_TO_USE}" -gravity Center -crop "${CROP_WIDTH}x${CROP_HEIGHT}+${CROP_OFFSET_X}+${CROP_OFFSET_Y}" +repage "${IMAGE_TO_USE}"
+	if [ $? -ne 0 ] ; then
+		echo -e "${RED}*** ${ME}: ERROR: CROP_IMAGE failed; not saving{$NC}"
+		exit 4
+	fi
+fi
+
+# Stretch the image if required, but only at night.
+if [ "${DAY_OR_NIGHT}" = "NIGHT" -a ${AUTO_STRETCH} = "true" ]; then
+	[ "${ALLSKY_DEBUG_LEVEL}" -ge 4 ] && echo "${ME}: Stretching '${IMAGE_TO_USE}' by ${AUTO_STRETCH_AMOUNT}"
+ 	convert "${IMAGE_TO_USE}" -sigmoidal-contrast "${AUTO_STRETCH_AMOUNT},${AUTO_STRETCH_MID_POINT}" "${IMAGE_TO_USE}"
+	if [ $? -ne 0 ] ; then
+		echo -e "${RED}*** ${ME}: ERROR: AUTO_STRETCH failed; not saving${NC}"
+		exit 4
+	fi
+fi
+
+if [ "${DAYTIME_SAVE}" = "true" -o "${DAY_OR_NIGHT}" = "NIGHT" ] ; then
+	# Determine what directory is the final resting place.
+	if [ "${DAY_OR_NIGHT}" = "NIGHT" ] ; then
+		# The 12 hours ago option ensures that we're always using today's date
+		# even at high latitudes where civil twilight can start after midnight.
+		DATE_DIR="${ALLSKY_IMAGES}/$(date -d '12 hours ago' +'%Y%m%d')"
+	else
+		# During the daytime we alway save the file in today's directory.
+		DATE_DIR="${ALLSKY_IMAGES}/$(date +'%Y%m%d')"
+	fi
+	mkdir -p "${DATE_DIR}"
+fi
+
+# If needed, save the current image in today's directory.
+if [ "${DAYTIME_SAVE}" = "true" -o "${DAY_OR_NIGHT}" = "NIGHT" ] ; then
```

Comment:
the same if statement ? (look a few lines higher)

Reply:
Ooops, good catch.